### DISCUSSION TITLE
Ragdoll Stops Moving.

### DISCUSSION THREAD

**Discussion by motherd** [jrouwe/JoltPhysics.js#191] [Q&A] [answered]
When a ragdoll is left in a state of little movement, the physics simulation stops, and even applying pressure with `Body.AddImpulse` does not make it move again.  
In my game, the number of ragdolls is small, so there would be no performance issue even if they didn't enter a sleep state.  
Is there a way to prevent the ragdolls from entering the sleep state, or to resume movement after they have stopped?

**Comment by jrouwe** [comment, marked as answer]
There are 2 ways:

1. If you use `BodyInterface::AddImpulse` instead of `Body::AddImpulse` then the body will be woken up automatically if it went to sleep (reason is that the body has no knowledge of the simulation it is part of, so it cannot trigger the wake up).
2. If you create the bodies with `BodyCreationSettings::mAllowSleeping = false` (through `RagdollSettings::mParts`) then they will never go to sleep

**Reply by motherd** [reply]
Thank you. The body no longer goes to sleep with `BodyCreationSettings::mAllowSleeping = false`.
